citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -494,4 +513,76 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
     public void deleteLocalGroupCitations(String groupId) {
         citationStore.deleteCitationByGroupId(groupId);
     }
+    
+    @Override
+    public void reprocessFile(IUser user, String zoteroGroupId, String itemId, String documentId)
+            throws GroupDoesNotExistException, CannotFindCitationException, ZoteroHttpStatusException,
+            ZoteroConnectionException, CitationIsOutdatedException, ZoteroItemCreationFailedException {
+        ICitation citation = getCitation(user, zoteroGroupId, itemId);
+        List<IGilesUpload> uploadsToReprocess = citation.getGilesUploads().stream()
+        .filter(upload -> upload.getDocumentId() != null && upload.getDocumentId().equals(documentId)).collect(Collectors.toList());
+        for(IGilesUpload upload : uploadsToReprocess) {
+            initiateReprocessing(user, documentId, citation);
+        }
+    }
+    
+    private void initiateReprocessing(IUser user, String documentId, ICitation citation) {
+        ResponseEntity<String> reprocessingResponse = gilesConnector.reprocessDocument(user, documentId);
+        if (reprocessingResponse.getStatusCode().equals(HttpStatus.OK)) {
+            IGilesUpload reprocessedUpload = new GilesUpload();
+            String responseBody = reprocessingResponse.getBody();
+            ObjectMapper objectMapper = new ObjectMapper();
+            String progressId = null;
+            try {
+                JsonNode jsonNode = objectMapper.readTree(responseBody);
+                progressId = jsonNode.get("id").asText();
+            } catch (IOException e) {
+                logger.error("Could not deserialize response.", e);
```

Comment:
this should probably not simply be ignored but the user needs to know. the rest doesn't really make sense if this fails, does it?